src/components/data-exploration/hierarchical-tree/HierarchicalTree.jsx

```diff
@@ -278,16 +274,27 @@ const HierarchicalTree = (props) => {
     return toRender;
   };
 
-  const treeDataToRender = renderTitlesRecursive(treeData);
+  const [renderedTreeData, setRenderedTreeData] = useState([]);
+  useEffect(() => {
+    if (!treeData) {
+      return;
+    }
+
+    setRenderedTreeData(renderTitlesRecursive(treeData));
+  }, [treeData]);
+
+  if (!treeData) return <Skeleton active />;
```

Comment:
if there is no treeData, shouldn't this component be an error state, not loading state?

Reply:
Again, no. In that case `CellSetsTool` would pass an empty array, and not a falsy value.